examples/compute/main.c

```diff
@@ -0,0 +1,180 @@
+#include "./../../ffi/wgpu.h"
+#include <stdio.h>
+#include <stdlib.h>
+
+#define BINDINGS_LENGTH (1)
+#define BIND_GROUP_LAYOUTS_LENGTH (1)
+
+WGPUByteArray read_file(const char *name) {
+    FILE *file = fopen(name, "rb");
+    fseek(file, 0, SEEK_END);
+    long length = ftell(file);
+    unsigned char *bytes = malloc(length);
+    fseek(file, 0, SEEK_SET);
+    fread(bytes, 1, length, file);
+    fclose(file);
+    return (WGPUByteArray){
+        .bytes = bytes,
+        .length = length,
+    };
+}
+
+void read_buffer_map(
+    WGPUBufferMapAsyncStatus status, 
+    const uint8_t *data, 
+    uint8_t *userdata) {
+    if (status == WGPUBufferMapAsyncStatus_Success) {
+        uint32_t *times = (uint32_t *) data;
+        printf("Times: [%d, %d, %d, %d]",
+            times[0], 
+            times[1], 
+            times[2], 
+            times[3]);
+    }
+}
+
+int main(
+    int argc, 
+    char *argv[]) {
+
+    if (argc != 5) {
```

Comment:
why can't we support arbitrary number of inputs?

Reply:
I always believed we only wanted to input four numbers for the example. Will change this to accept a random number of inputs.

Reply:
Yeah I think the original Rust example might work this way (we could change both though)

Reply:
Just checked out both gfx and wgpu-rs, both support arbitrary number of inputs. I will change this accordingly.

Reply:
Tried to make this work, but I cannot create a variable length array, because we are using C89. I don't know if we want to use a newer version of C, like C99. So for now I will force the user to only pass 4 integers.